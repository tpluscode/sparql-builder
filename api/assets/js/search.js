var typedoc = typedoc || {};
            typedoc.search = typedoc.search || {};
            typedoc.search.data = {"kinds":{"1":"Module","32":"Variable","64":"Function","256":"Interface","1024":"Property","2048":"Method","65536":"Type literal","2097152":"Object literal","4194304":"Type alias","16777216":"Reference"},"rows":[{"id":0,"kind":1,"name":"\"lib/index\"","url":"modules/_lib_index_.html","classes":"tsd-kind-module"},{"id":1,"kind":256,"name":"SparqlBuildOptions","url":"interfaces/_lib_index_.sparqlbuildoptions.html","classes":"tsd-kind-interface tsd-parent-kind-module tsd-is-not-exported","parent":"\"lib/index\""},{"id":2,"kind":1024,"name":"base","url":"interfaces/_lib_index_.sparqlbuildoptions.html#base","classes":"tsd-kind-property tsd-parent-kind-interface tsd-is-not-exported","parent":"\"lib/index\".SparqlBuildOptions"},{"id":3,"kind":256,"name":"SparqlQuery","url":"interfaces/_lib_index_.sparqlquery.html","classes":"tsd-kind-interface tsd-parent-kind-module","parent":"\"lib/index\""},{"id":4,"kind":2048,"name":"build","url":"interfaces/_lib_index_.sparqlquery.html#build","classes":"tsd-kind-method tsd-parent-kind-interface","parent":"\"lib/index\".SparqlQuery"},{"id":5,"kind":2048,"name":"_getTemplateResult","url":"interfaces/_lib_index_.sparqlquery.html#_gettemplateresult","classes":"tsd-kind-method tsd-parent-kind-interface","parent":"\"lib/index\".SparqlQuery"},{"id":6,"kind":256,"name":"SparqlQueryExecutable","url":"interfaces/_lib_index_.sparqlqueryexecutable.html","classes":"tsd-kind-interface tsd-parent-kind-module","parent":"\"lib/index\""},{"id":7,"kind":2048,"name":"execute","url":"interfaces/_lib_index_.sparqlqueryexecutable.html#execute","classes":"tsd-kind-method tsd-parent-kind-interface tsd-has-type-parameter","parent":"\"lib/index\".SparqlQueryExecutable"},{"id":8,"kind":256,"name":"SparqlGraphQueryExecutable","url":"interfaces/_lib_index_.sparqlgraphqueryexecutable.html","classes":"tsd-kind-interface tsd-parent-kind-module","parent":"\"lib/index\""},{"id":9,"kind":2048,"name":"execute","url":"interfaces/_lib_index_.sparqlgraphqueryexecutable.html#execute","classes":"tsd-kind-method tsd-parent-kind-interface tsd-has-type-parameter","parent":"\"lib/index\".SparqlGraphQueryExecutable"},{"id":10,"kind":256,"name":"SparqlUpdateExecutable","url":"interfaces/_lib_index_.sparqlupdateexecutable.html","classes":"tsd-kind-interface tsd-parent-kind-module","parent":"\"lib/index\""},{"id":11,"kind":2048,"name":"execute","url":"interfaces/_lib_index_.sparqlupdateexecutable.html#execute","classes":"tsd-kind-method tsd-parent-kind-interface tsd-has-type-parameter","parent":"\"lib/index\".SparqlUpdateExecutable"},{"id":12,"kind":256,"name":"SparqlAskExecutable","url":"interfaces/_lib_index_.sparqlaskexecutable.html","classes":"tsd-kind-interface tsd-parent-kind-module","parent":"\"lib/index\""},{"id":13,"kind":2048,"name":"execute","url":"interfaces/_lib_index_.sparqlaskexecutable.html#execute","classes":"tsd-kind-method tsd-parent-kind-interface tsd-has-type-parameter","parent":"\"lib/index\".SparqlAskExecutable"},{"id":14,"kind":4194304,"name":"SparqlExecuteOptions","url":"modules/_lib_index_.html#sparqlexecuteoptions","classes":"tsd-kind-type-alias tsd-parent-kind-module","parent":"\"lib/index\""},{"id":15,"kind":4194304,"name":"Builder","url":"modules/_lib_index_.html#builder","classes":"tsd-kind-type-alias tsd-parent-kind-module tsd-is-not-exported","parent":"\"lib/index\""},{"id":16,"kind":64,"name":"Builder","url":"modules/_lib_index_.html#builder-1","classes":"tsd-kind-function tsd-parent-kind-module tsd-has-type-parameter","parent":"\"lib/index\""},{"id":17,"kind":1,"name":"\"lib/execute\"","url":"modules/_lib_execute_.html","classes":"tsd-kind-module"},{"id":18,"kind":32,"name":"logQuery","url":"modules/_lib_execute_.html#logquery","classes":"tsd-kind-variable tsd-parent-kind-module tsd-is-not-exported","parent":"\"lib/execute\""},{"id":19,"kind":32,"name":"logQueryError","url":"modules/_lib_execute_.html#logqueryerror","classes":"tsd-kind-variable tsd-parent-kind-module tsd-is-not-exported","parent":"\"lib/execute\""},{"id":20,"kind":64,"name":"buildAndRun","url":"modules/_lib_execute_.html#buildandrun","classes":"tsd-kind-function tsd-parent-kind-module tsd-has-type-parameter tsd-is-not-exported","parent":"\"lib/execute\""},{"id":21,"kind":2097152,"name":"update","url":"modules/_lib_execute_.html#update","classes":"tsd-kind-object-literal tsd-parent-kind-module","parent":"\"lib/execute\""},{"id":22,"kind":64,"name":"execute","url":"modules/_lib_execute_.html#update.execute-3","classes":"tsd-kind-function tsd-parent-kind-object-literal tsd-has-type-parameter","parent":"\"lib/execute\".update"},{"id":23,"kind":2097152,"name":"ask","url":"modules/_lib_execute_.html#ask","classes":"tsd-kind-object-literal tsd-parent-kind-module","parent":"\"lib/execute\""},{"id":24,"kind":64,"name":"execute","url":"modules/_lib_execute_.html#ask.execute","classes":"tsd-kind-function tsd-parent-kind-object-literal tsd-has-type-parameter","parent":"\"lib/execute\".ask"},{"id":25,"kind":2097152,"name":"select","url":"modules/_lib_execute_.html#select","classes":"tsd-kind-object-literal tsd-parent-kind-module","parent":"\"lib/execute\""},{"id":26,"kind":64,"name":"execute","url":"modules/_lib_execute_.html#select.execute-2","classes":"tsd-kind-function tsd-parent-kind-object-literal tsd-has-type-parameter","parent":"\"lib/execute\".select"},{"id":27,"kind":2097152,"name":"graph","url":"modules/_lib_execute_.html#graph","classes":"tsd-kind-object-literal tsd-parent-kind-module","parent":"\"lib/execute\""},{"id":28,"kind":64,"name":"execute","url":"modules/_lib_execute_.html#graph.execute-1","classes":"tsd-kind-function tsd-parent-kind-object-literal tsd-has-type-parameter","parent":"\"lib/execute\".graph"},{"id":29,"kind":1,"name":"\"lib/TemplateResult\"","url":"modules/_lib_templateresult_.html","classes":"tsd-kind-module"},{"id":30,"kind":64,"name":"concat","url":"modules/_lib_templateresult_.html#concat","classes":"tsd-kind-function tsd-parent-kind-module","parent":"\"lib/TemplateResult\""},{"id":31,"kind":1,"name":"\"lib/partials/DATA\"","url":"modules/_lib_partials_data_.html","classes":"tsd-kind-module"},{"id":32,"kind":256,"name":"QuadDataBuilder","url":"interfaces/_lib_partials_data_.quaddatabuilder.html","classes":"tsd-kind-interface tsd-parent-kind-module tsd-has-type-parameter","parent":"\"lib/partials/DATA\""},{"id":33,"kind":1024,"name":"quadData","url":"interfaces/_lib_partials_data_.quaddatabuilder.html#quaddata","classes":"tsd-kind-property tsd-parent-kind-interface","parent":"\"lib/partials/DATA\".QuadDataBuilder"},{"id":34,"kind":2048,"name":"DATA","url":"interfaces/_lib_partials_data_.quaddatabuilder.html#data","classes":"tsd-kind-method tsd-parent-kind-interface","parent":"\"lib/partials/DATA\".QuadDataBuilder"},{"id":35,"kind":1,"name":"\"lib/partials/WHERE\"","url":"modules/_lib_partials_where_.html","classes":"tsd-kind-module"},{"id":36,"kind":256,"name":"WhereBuilder","url":"interfaces/_lib_partials_where_.wherebuilder.html","classes":"tsd-kind-interface tsd-parent-kind-module tsd-has-type-parameter","parent":"\"lib/partials/WHERE\""},{"id":37,"kind":1024,"name":"patterns","url":"interfaces/_lib_partials_where_.wherebuilder.html#patterns","classes":"tsd-kind-property tsd-parent-kind-interface","parent":"\"lib/partials/WHERE\".WhereBuilder"},{"id":38,"kind":2048,"name":"whereClause","url":"interfaces/_lib_partials_where_.wherebuilder.html#whereclause","classes":"tsd-kind-method tsd-parent-kind-interface","parent":"\"lib/partials/WHERE\".WhereBuilder"},{"id":39,"kind":2048,"name":"WHERE","url":"interfaces/_lib_partials_where_.wherebuilder.html#where","classes":"tsd-kind-method tsd-parent-kind-interface","parent":"\"lib/partials/WHERE\".WhereBuilder"},{"id":40,"kind":1,"name":"\"lib/partials/INSERT\"","url":"modules/_lib_partials_insert_.html","classes":"tsd-kind-module"},{"id":41,"kind":256,"name":"InsertBuilder","url":"interfaces/_lib_partials_insert_.insertbuilder.html","classes":"tsd-kind-interface tsd-parent-kind-module tsd-has-type-parameter","parent":"\"lib/partials/INSERT\""},{"id":42,"kind":1024,"name":"insertPatterns","url":"interfaces/_lib_partials_insert_.insertbuilder.html#insertpatterns","classes":"tsd-kind-property tsd-parent-kind-interface","parent":"\"lib/partials/INSERT\".InsertBuilder"},{"id":43,"kind":2048,"name":"insertClause","url":"interfaces/_lib_partials_insert_.insertbuilder.html#insertclause","classes":"tsd-kind-method tsd-parent-kind-interface","parent":"\"lib/partials/INSERT\".InsertBuilder"},{"id":44,"kind":2048,"name":"INSERT","url":"interfaces/_lib_partials_insert_.insertbuilder.html#insert","classes":"tsd-kind-method tsd-parent-kind-interface","parent":"\"lib/partials/INSERT\".InsertBuilder"},{"id":45,"kind":1,"name":"\"lib/DeleteBuilder\"","url":"modules/_lib_deletebuilder_.html","classes":"tsd-kind-module"},{"id":46,"kind":4194304,"name":"DeleteInsertQuery","url":"modules/_lib_deletebuilder_.html#deleteinsertquery","classes":"tsd-kind-type-alias tsd-parent-kind-module","parent":"\"lib/DeleteBuilder\""},{"id":47,"kind":4194304,"name":"DeleteData","url":"modules/_lib_deletebuilder_.html#deletedata","classes":"tsd-kind-type-alias tsd-parent-kind-module tsd-is-not-exported","parent":"\"lib/DeleteBuilder\""},{"id":48,"kind":64,"name":"DELETE","url":"modules/_lib_deletebuilder_.html#delete","classes":"tsd-kind-function tsd-parent-kind-module","parent":"\"lib/DeleteBuilder\""},{"id":49,"kind":1,"name":"\"lib/InsertBuilder\"","url":"modules/_lib_insertbuilder_.html","classes":"tsd-kind-module"},{"id":50,"kind":4194304,"name":"InsertQuery","url":"modules/_lib_insertbuilder_.html#insertquery","classes":"tsd-kind-type-alias tsd-parent-kind-module","parent":"\"lib/InsertBuilder\""},{"id":51,"kind":4194304,"name":"InsertData","url":"modules/_lib_insertbuilder_.html#insertdata","classes":"tsd-kind-type-alias tsd-parent-kind-module tsd-is-not-exported","parent":"\"lib/InsertBuilder\""},{"id":52,"kind":64,"name":"INSERT","url":"modules/_lib_insertbuilder_.html#insert","classes":"tsd-kind-function tsd-parent-kind-module","parent":"\"lib/InsertBuilder\""},{"id":53,"kind":1,"name":"\"lib/partials/LIMIT\"","url":"modules/_lib_partials_limit_.html","classes":"tsd-kind-module"},{"id":54,"kind":256,"name":"LimitOffsetBuilder","url":"interfaces/_lib_partials_limit_.limitoffsetbuilder.html","classes":"tsd-kind-interface tsd-parent-kind-module tsd-has-type-parameter","parent":"\"lib/partials/LIMIT\""},{"id":55,"kind":1024,"name":"offset","url":"interfaces/_lib_partials_limit_.limitoffsetbuilder.html#offset","classes":"tsd-kind-property tsd-parent-kind-interface","parent":"\"lib/partials/LIMIT\".LimitOffsetBuilder"},{"id":56,"kind":1024,"name":"limit","url":"interfaces/_lib_partials_limit_.limitoffsetbuilder.html#limit","classes":"tsd-kind-property tsd-parent-kind-interface","parent":"\"lib/partials/LIMIT\".LimitOffsetBuilder"},{"id":57,"kind":2048,"name":"LIMIT","url":"interfaces/_lib_partials_limit_.limitoffsetbuilder.html#limit-1","classes":"tsd-kind-method tsd-parent-kind-interface","parent":"\"lib/partials/LIMIT\".LimitOffsetBuilder"},{"id":58,"kind":2048,"name":"OFFSET","url":"interfaces/_lib_partials_limit_.limitoffsetbuilder.html#offset-1","classes":"tsd-kind-method tsd-parent-kind-interface","parent":"\"lib/partials/LIMIT\".LimitOffsetBuilder"},{"id":59,"kind":2048,"name":"limitOffsetClause","url":"interfaces/_lib_partials_limit_.limitoffsetbuilder.html#limitoffsetclause","classes":"tsd-kind-method tsd-parent-kind-interface","parent":"\"lib/partials/LIMIT\".LimitOffsetBuilder"},{"id":60,"kind":1,"name":"\"lib/partials/FROM\"","url":"modules/_lib_partials_from_.html","classes":"tsd-kind-module"},{"id":61,"kind":256,"name":"FromNamed","url":"interfaces/_lib_partials_from_.fromnamed.html","classes":"tsd-kind-interface tsd-parent-kind-module tsd-has-type-parameter tsd-is-not-exported","parent":"\"lib/partials/FROM\""},{"id":62,"kind":2048,"name":"NAMED","url":"interfaces/_lib_partials_from_.fromnamed.html#named","classes":"tsd-kind-method tsd-parent-kind-interface tsd-is-not-exported","parent":"\"lib/partials/FROM\".FromNamed"},{"id":63,"kind":256,"name":"FromBuilder","url":"interfaces/_lib_partials_from_.frombuilder.html","classes":"tsd-kind-interface tsd-parent-kind-module tsd-has-type-parameter","parent":"\"lib/partials/FROM\""},{"id":64,"kind":1024,"name":"defaultGraph","url":"interfaces/_lib_partials_from_.frombuilder.html#defaultgraph","classes":"tsd-kind-property tsd-parent-kind-interface","parent":"\"lib/partials/FROM\".FromBuilder"},{"id":65,"kind":1024,"name":"fromNamed","url":"interfaces/_lib_partials_from_.frombuilder.html#fromnamed","classes":"tsd-kind-property tsd-parent-kind-interface","parent":"\"lib/partials/FROM\".FromBuilder"},{"id":66,"kind":2048,"name":"FROM","url":"interfaces/_lib_partials_from_.frombuilder.html#from","classes":"tsd-kind-method tsd-parent-kind-interface","parent":"\"lib/partials/FROM\".FromBuilder"},{"id":67,"kind":2048,"name":"fromClause","url":"interfaces/_lib_partials_from_.frombuilder.html#fromclause","classes":"tsd-kind-method tsd-parent-kind-interface","parent":"\"lib/partials/FROM\".FromBuilder"},{"id":68,"kind":64,"name":"default","url":"modules/_lib_partials_from_.html#default","classes":"tsd-kind-function tsd-parent-kind-module tsd-has-type-parameter tsd-is-not-exported","parent":"\"lib/partials/FROM\""},{"id":69,"kind":1,"name":"\"lib/AskBuilder\"","url":"modules/_lib_askbuilder_.html","classes":"tsd-kind-module"},{"id":70,"kind":4194304,"name":"AskQuery","url":"modules/_lib_askbuilder_.html#askquery","classes":"tsd-kind-type-alias tsd-parent-kind-module tsd-is-not-exported","parent":"\"lib/AskBuilder\""},{"id":71,"kind":64,"name":"ASK","url":"modules/_lib_askbuilder_.html#ask","classes":"tsd-kind-function tsd-parent-kind-module","parent":"\"lib/AskBuilder\""},{"id":72,"kind":1,"name":"\"lib/DescribeBuilder\"","url":"modules/_lib_describebuilder_.html","classes":"tsd-kind-module"},{"id":73,"kind":4194304,"name":"DescribeQuery","url":"modules/_lib_describebuilder_.html#describequery","classes":"tsd-kind-type-alias tsd-parent-kind-module tsd-is-not-exported","parent":"\"lib/DescribeBuilder\""},{"id":74,"kind":64,"name":"DESCRIBE","url":"modules/_lib_describebuilder_.html#describe","classes":"tsd-kind-function tsd-parent-kind-module","parent":"\"lib/DescribeBuilder\""},{"id":75,"kind":1,"name":"\"lib/ConstructBuilder\"","url":"modules/_lib_constructbuilder_.html","classes":"tsd-kind-module"},{"id":76,"kind":4194304,"name":"ConstructQuery","url":"modules/_lib_constructbuilder_.html#constructquery","classes":"tsd-kind-type-alias tsd-parent-kind-module tsd-is-not-exported","parent":"\"lib/ConstructBuilder\""},{"id":77,"kind":64,"name":"CONSTRUCT","url":"modules/_lib_constructbuilder_.html#construct","classes":"tsd-kind-function tsd-parent-kind-module","parent":"\"lib/ConstructBuilder\""},{"id":78,"kind":1,"name":"\"lib/partials/ORDER\"","url":"modules/_lib_partials_order_.html","classes":"tsd-kind-module"},{"id":79,"kind":256,"name":"OrderCondition","url":"interfaces/_lib_partials_order_.ordercondition.html","classes":"tsd-kind-interface tsd-parent-kind-module tsd-is-not-exported","parent":"\"lib/partials/ORDER\""},{"id":80,"kind":1024,"name":"variable","url":"interfaces/_lib_partials_order_.ordercondition.html#variable","classes":"tsd-kind-property tsd-parent-kind-interface tsd-is-not-exported","parent":"\"lib/partials/ORDER\".OrderCondition"},{"id":81,"kind":1024,"name":"desc","url":"interfaces/_lib_partials_order_.ordercondition.html#desc","classes":"tsd-kind-property tsd-parent-kind-interface tsd-is-not-exported","parent":"\"lib/partials/ORDER\".OrderCondition"},{"id":82,"kind":256,"name":"ThenOrderByBuilder","url":"interfaces/_lib_partials_order_.thenorderbybuilder.html","classes":"tsd-kind-interface tsd-parent-kind-module tsd-has-type-parameter tsd-is-not-exported","parent":"\"lib/partials/ORDER\""},{"id":83,"kind":1024,"name":"THEN","url":"interfaces/_lib_partials_order_.thenorderbybuilder.html#then","classes":"tsd-kind-property tsd-parent-kind-interface tsd-is-not-exported","parent":"\"lib/partials/ORDER\".ThenOrderByBuilder"},{"id":84,"kind":256,"name":"OrderByBuilder","url":"interfaces/_lib_partials_order_.orderbybuilder.html","classes":"tsd-kind-interface tsd-parent-kind-module tsd-has-type-parameter tsd-is-not-exported","parent":"\"lib/partials/ORDER\""},{"id":85,"kind":2048,"name":"BY","url":"interfaces/_lib_partials_order_.orderbybuilder.html#by","classes":"tsd-kind-method tsd-parent-kind-interface tsd-is-not-exported","parent":"\"lib/partials/ORDER\".OrderByBuilder"},{"id":86,"kind":256,"name":"OrderBuilder","url":"interfaces/_lib_partials_order_.orderbuilder.html","classes":"tsd-kind-interface tsd-parent-kind-module tsd-has-type-parameter","parent":"\"lib/partials/ORDER\""},{"id":87,"kind":1024,"name":"orderConditions","url":"interfaces/_lib_partials_order_.orderbuilder.html#orderconditions","classes":"tsd-kind-property tsd-parent-kind-interface","parent":"\"lib/partials/ORDER\".OrderBuilder"},{"id":88,"kind":2048,"name":"orderClause","url":"interfaces/_lib_partials_order_.orderbuilder.html#orderclause","classes":"tsd-kind-method tsd-parent-kind-interface","parent":"\"lib/partials/ORDER\".OrderBuilder"},{"id":89,"kind":2048,"name":"ORDER","url":"interfaces/_lib_partials_order_.orderbuilder.html#order","classes":"tsd-kind-method tsd-parent-kind-interface","parent":"\"lib/partials/ORDER\".OrderBuilder"},{"id":90,"kind":64,"name":"addOrder","url":"modules/_lib_partials_order_.html#addorder","classes":"tsd-kind-function tsd-parent-kind-module tsd-has-type-parameter tsd-is-not-exported","parent":"\"lib/partials/ORDER\""},{"id":91,"kind":1,"name":"\"lib/SelectBuilder\"","url":"modules/_lib_selectbuilder_.html","classes":"tsd-kind-module"},{"id":92,"kind":256,"name":"Select","url":"interfaces/_lib_selectbuilder_.select.html","classes":"tsd-kind-interface tsd-parent-kind-module tsd-is-not-exported","parent":"\"lib/SelectBuilder\""},{"id":93,"kind":1024,"name":"DISTINCT","url":"interfaces/_lib_selectbuilder_.select.html#distinct","classes":"tsd-kind-property tsd-parent-kind-interface tsd-is-not-exported","parent":"\"lib/SelectBuilder\".Select"},{"id":94,"kind":65536,"name":"__type","url":"interfaces/_lib_selectbuilder_.select.html#distinct.__type","classes":"tsd-kind-type-literal tsd-parent-kind-property tsd-is-not-exported","parent":"\"lib/SelectBuilder\".Select.DISTINCT"},{"id":95,"kind":1024,"name":"REDUCED","url":"interfaces/_lib_selectbuilder_.select.html#reduced","classes":"tsd-kind-property tsd-parent-kind-interface tsd-is-not-exported","parent":"\"lib/SelectBuilder\".Select"},{"id":96,"kind":65536,"name":"__type","url":"interfaces/_lib_selectbuilder_.select.html#reduced.__type-1","classes":"tsd-kind-type-literal tsd-parent-kind-property tsd-is-not-exported","parent":"\"lib/SelectBuilder\".Select.REDUCED"},{"id":97,"kind":1024,"name":"ALL","url":"interfaces/_lib_selectbuilder_.select.html#all","classes":"tsd-kind-property tsd-parent-kind-interface tsd-is-not-exported","parent":"\"lib/SelectBuilder\".Select"},{"id":98,"kind":4194304,"name":"SelectQuery","url":"modules/_lib_selectbuilder_.html#selectquery","classes":"tsd-kind-type-alias tsd-parent-kind-module tsd-is-not-exported","parent":"\"lib/SelectBuilder\""},{"id":99,"kind":64,"name":"SelectBuilder","url":"modules/_lib_selectbuilder_.html#selectbuilder","classes":"tsd-kind-function tsd-parent-kind-module tsd-is-not-exported","parent":"\"lib/SelectBuilder\""},{"id":100,"kind":32,"name":"SELECT","url":"modules/_lib_selectbuilder_.html#select-1","classes":"tsd-kind-variable tsd-parent-kind-module","parent":"\"lib/SelectBuilder\""},{"id":101,"kind":1,"name":"\"lib/WithBuilder\"","url":"modules/_lib_withbuilder_.html","classes":"tsd-kind-module"},{"id":102,"kind":4194304,"name":"WithQuery","url":"modules/_lib_withbuilder_.html#withquery","classes":"tsd-kind-type-alias tsd-parent-kind-module tsd-is-not-exported","parent":"\"lib/WithBuilder\""},{"id":103,"kind":64,"name":"WITH","url":"modules/_lib_withbuilder_.html#with","classes":"tsd-kind-function tsd-parent-kind-module","parent":"\"lib/WithBuilder\""},{"id":104,"kind":1,"name":"\"index\"","url":"modules/_index_.html","classes":"tsd-kind-module"},{"id":105,"kind":16777216,"name":"sparql","url":"modules/_index_.html#sparql","classes":"tsd-kind-reference tsd-parent-kind-module","parent":"\"index\""},{"id":106,"kind":16777216,"name":"DELETE","url":"modules/_index_.html#delete","classes":"tsd-kind-reference tsd-parent-kind-module","parent":"\"index\""},{"id":107,"kind":16777216,"name":"INSERT","url":"modules/_index_.html#insert","classes":"tsd-kind-reference tsd-parent-kind-module","parent":"\"index\""},{"id":108,"kind":16777216,"name":"ASK","url":"modules/_index_.html#ask","classes":"tsd-kind-reference tsd-parent-kind-module","parent":"\"index\""},{"id":109,"kind":16777216,"name":"DESCRIBE","url":"modules/_index_.html#describe","classes":"tsd-kind-reference tsd-parent-kind-module","parent":"\"index\""},{"id":110,"kind":16777216,"name":"CONSTRUCT","url":"modules/_index_.html#construct","classes":"tsd-kind-reference tsd-parent-kind-module","parent":"\"index\""},{"id":111,"kind":16777216,"name":"SELECT","url":"modules/_index_.html#select","classes":"tsd-kind-reference tsd-parent-kind-module","parent":"\"index\""},{"id":112,"kind":16777216,"name":"WITH","url":"modules/_index_.html#with","classes":"tsd-kind-reference tsd-parent-kind-module","parent":"\"index\""}]};