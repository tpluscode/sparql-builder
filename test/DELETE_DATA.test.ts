import RDF from '@zazuko/env'
import chai, { expect } from 'chai'
import { jestSnapshotPlugin } from 'mocha-chai-jest-snapshot'
import { DELETE } from '../src/index.js'
import { sparqlClient } from './_mocks.js'
import './sparql.js'

describe('DELETE DATA', () => {
  chai.use(jestSnapshotPlugin())

  it('builds correct query', () => {
    // given
    const expected = `PREFIX owl: <http://www.w3.org/2002/07/owl#>
DELETE DATA { <http://example.com> owl:sameAs <http://example.org> . }`

    // when
    const actual = DELETE.DATA`<http://example.com> ${RDF.ns.owl.sameAs} <http://example.org>`.build()

    // then
    expect(actual).to.be.query(expected)
  })

  it('executes as update', async () => {
    // given
    const client = sparqlClient()

    // when
    await DELETE.DATA`<http://example.com> owl:sameAs <http://example.org>`.execute(client)

    // then
    expect(client.query.update).to.have.been.called
  })

  it('can have additional prologue', function () {
    // given
    const base = RDF.namedNode('http://foo.bar/baz')

    // when
    const query = DELETE.DATA`<http://example.com/bar> ${RDF.ns.owl.sameAs} <http://example.org/bar> .`
      .prologue`#pragma join.hash off`
      .prologue`BASE ${base}`
      .build()

    // then
    expect(query).toMatchSnapshot()
  })

  it('can delete triples', function () {
    // given
    const data = RDF.quad(
      RDF.namedNode('http://example.com/bar'),
      RDF.ns.owl.sameAs,
      RDF.namedNode('http://example.com/bar'),
    )

    // when
    const query = DELETE.DATA`${data}`.build()

    // then
    expect(query).toMatchSnapshot()
  })

  it('can delete quads', function () {
    // given
    const data = RDF.quad(
      RDF.namedNode('http://example.com/bar'),
      RDF.ns.owl.sameAs,
      RDF.namedNode('http://example.com/bar'),
      RDF.namedNode('http://example.com/G'),
    )

    // when
    const query = DELETE.DATA`${data}`.build()

    // then
    expect(query).toMatchSnapshot()
  })

  it('can delete dataset', function () {
    // given
    const data = RDF.dataset([RDF.quad(
      RDF.namedNode('http://example.com/bar'),
      RDF.ns.owl.sameAs,
      RDF.namedNode('http://example.com/bar'),
    )])

    // when
    const query = DELETE.DATA`${data}`.build()

    // then
    expect(query).toMatchSnapshot()
  })

  it('can chain multiple quad data calls', () => {
    // given
    const expected = `PREFIX owl: <http://www.w3.org/2002/07/owl#>
DELETE DATA { 
    <http://example.com/bar> owl:sameAs <http://example.org/bar> .
    <http://example.com/foo> owl:sameAs <http://example.org/foo> .
}`

    // when
    const actual = DELETE
      .DATA`<http://example.com/bar> ${RDF.ns.owl.sameAs} <http://example.org/bar> .`
      .DATA`<http://example.com/foo> ${RDF.ns.owl.sameAs} <http://example.org/foo> .`
      .build()

    // then
    expect(actual).to.be.query(expected)
  })
})
