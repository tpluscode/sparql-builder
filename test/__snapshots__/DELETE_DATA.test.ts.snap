// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DELETE DATA can delete dataset 1`] = `
"PREFIX owl: <http://www.w3.org/2002/07/owl#>

DELETE DATA {
  

<http://example.com/bar> owl:sameAs <http://example.com/bar> .
}"
`;

exports[`DELETE DATA can delete quads 1`] = `
"PREFIX owl: <http://www.w3.org/2002/07/owl#>

DELETE DATA {
  GRAPH <http://example.com/G> { <http://example.com/bar> owl:sameAs <http://example.com/bar> . }
}"
`;

exports[`DELETE DATA can delete triples 1`] = `
"PREFIX owl: <http://www.w3.org/2002/07/owl#>

DELETE DATA {
  <http://example.com/bar> owl:sameAs <http://example.com/bar> .
}"
`;

exports[`DELETE DATA can have additional prologue 1`] = `
"#pragma join.hash off
BASE <http://foo.bar/baz>

PREFIX owl: <http://www.w3.org/2002/07/owl#>

DELETE DATA {
  <http://example.com/bar> owl:sameAs <http://example.org/bar> .
}"
`;
